apis/risingwave/v1alpha1/risingwave_standalone_mode.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RisingWaveStandaloneComponent contains the spec for standalone component.
+type RisingWaveStandaloneComponent struct {
+	// LogLevel controls the log level of the running nodes. It can be in any format that the underlying component supports,
+	// e.g., in the RUST_LOG format for Rust programs. Defaults to INFO.
+	// +kubebuilder:default=INFO
+	LogLevel string `json:"logLevel,omitempty"`
+
+	// DisallowPrintStackTraces determines if the stack traces are allowed to print when panic happens. This options applies
+	// to both Rust and Java programs. Defaults to false.
+	// +optional
+	DisallowPrintStackTraces *bool `json:"disallowPrintStackTraces,omitempty"`
+
+	// RestartAt is the time that the Pods under the group should be restarted. Setting a value on this field will
+	// trigger a full recreation of the Pods. Defaults to nil.
+	RestartAt *metav1.Time `json:"restartAt,omitempty"`
+
+	// Replicas is the number of the standalone Pods. Maximum is 1. Defaults to 1.
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=1
+	Replicas int32 `json:"replicas,omitempty"`
```

Comment:
What will happend if user choose standalone but Replicas is 0?

Reply:
Then the RisingWave is paused. Just like the replicas fields are 0 when in clustered deployment.